alpine/ExamplesWithoutPicManager/LandauDampingParameterList.cpp

```diff
@@ -0,0 +1,425 @@
+
+        using bunch_type = ChargedParticles<PLayout_t<double, Dim>, double, Dim>;
+
+        std::unique_ptr<bunch_type> P;
+
+        ippl::NDIndex<Dim> domain;
+        for (unsigned i = 0; i < Dim; i++) {
+            domain[i] = ippl::Index(nr[i]);
+        }
+
+        std::array<bool, Dim> isParallel;
+        isParallel.fill(true);
+
+        // create mesh and layout objects for this problem domain
+        Vector_t<double, Dim> kw = 0.5;
+        double alpha             = 0.05;
+        Vector_t<double, Dim> rmin(0.0);
+        Vector_t<double, Dim> rmax = 2 * pi / kw;
+
+        Vector_t<double, Dim> hr = rmax / nr;
+        // Q = -\int\int f dx dv
+        double Q = std::reduce(rmax.begin(), rmax.end(), -1., std::multiplies<double>());
+        Vector_t<double, Dim> origin = rmin;
+        const double dt              = std::min(.05, 0.5 * *std::min_element(hr.begin(), hr.end()));
+
+        const bool isAllPeriodic = true;
+        Mesh_t<Dim> mesh(domain, hr, origin);
+        FieldLayout_t<Dim> FL(MPI_COMM_WORLD, domain, isParallel, isAllPeriodic);
+        PLayout_t<double, Dim> PL(FL, mesh);
+
+        std::string solver = argv[arg++];
+        // MOD: Setup of Preconditioner Start
+        int gauss_seidel_inner_iterations;
+        int gauss_seidel_outer_iterations;
+        int newton_level;
+        int chebyshev_degree;
+        int richardson_iterations;
+        int communication;
+        std::string preconditioner_type = "";
+
+        if (solver == "OPEN") {
+            throw IpplException("LandauDamping",
+                                "Open boundaries solver incompatible with this simulation!");
+        }
+
+        if (solver == "PCG") {
+            preconditioner_type = argv[arg++];
+            if (preconditioner_type == "newton") {
+                newton_level = std::atoi(argv[arg++]);
+            } else if (preconditioner_type == "chebyshev") {
+                chebyshev_degree = std::atoi(argv[arg++]);
+            } else if (preconditioner_type == "richardson") {
+                richardson_iterations = std::atoi(argv[arg++]);
+                communication         = std::atoi(argv[arg++]);
+            } else if (preconditioner_type == "gauss-seidel") {
+                gauss_seidel_inner_iterations = std::atoi(argv[arg++]);
+                gauss_seidel_outer_iterations = std::atoi(argv[arg++]);
+                communication                 = std::atoi(argv[arg++]);
+            }
+        }
+
+        ippl::ParameterList params;
+        params.add("preconditioner_type", preconditioner_type);
+        params.add("gauss_seidel_inner_iterations", gauss_seidel_inner_iterations);
+        params.add("gauss_seidel_outer_iterations", gauss_seidel_outer_iterations);
+        params.add("newton_level", newton_level);
+        params.add("chebyshev_degree", chebyshev_degree);
+        params.add("richardson_iterations", richardson_iterations);
+        params.add("communication", communication);
```

Comment:
All these parameters are only necessary in the PCG solver case, so should they be inside the previous "if"? 
This is more of a design/correctness thing, since the parameters won't be used anyways if the solver is different.

Reply:
yes I agree they should be inside the if